compiler/rustc_const_eval/src/util/compare_types.rs

```diff
@@ -56,8 +56,13 @@ pub fn is_subtype<'tcx>(
     // With `Reveal::All`, opaque types get normalized away, with `Reveal::UserFacing`
     // we would get unification errors because we're unable to look into opaque types,
     // even if they're constrained in our current function.
-    //
-    // It seems very unlikely that this hides any bugs.
-    let _ = infcx.take_opaque_types();
+    for (key, ty) in infcx.take_opaque_types() {
+        span_bug!(
+            ty.hidden_type.span,
+            "{}, {}",
+            tcx.type_of(key.def_id).subst(tcx, key.substs),
+            ty.hidden_type.ty
+        );
```

Comment:
I feel like we should hit this bug? :thinking: or do we avoid it because we have `OpaqueCast` everywhere we would assign to/from opaques?

Reply:
I was surprised, too, but I didn't debug it.

Reply:
do we run the validator before optimizations by default :sweat_smile: iirc we only do it with `-Zvalidate-mir` :thinking:

Reply:
https://github.com/rust-lang/rust/issues/114121 :D